src/shared/bootspec.c

```diff
@@ -169,17 +169,122 @@ static int parse_path_many(
         return strv_extend_strv(s, f, /* filter_duplicates= */ false);
 }
 
+static int parse_tries(const char *fname, const char *p, size_t n, unsigned *ret) {
+        _cleanup_free_ char *d = NULL;
+        unsigned tries;
+        int r;
+
+        assert(fname);
+        assert(p);
+        assert(ret);
+
+        d = strndup(p, n);
+        if (!d)
+                return log_oom();
+
+        r = safe_atou(d, &tries);
+        if (r >= 0 && tries == UINT_MAX) /* We use UINT_MAX as internal marker for unset */
+                r = -ERANGE;
+        if (r < 0)
+                return log_error_errno(r, "Failed to parse tries counter of filename '%s': %m", fname);
+
+        *ret = tries;
+        return 0;
+}
+
+int boot_filename_extract_tries(
```

Comment:
This is a rather roundabout way of parsing this and rather verbose. See `config_entry_parse_tries` in `boot.c` how it can be done so much easier.

Reply:
I am not sure it's that much better... implements strrchr() manually even?

Reply:
`strrchr` on `+` and 2 calls to `safe_atou_full` does not strike you as simpler than all this manual parsing, huh?

Reply:
hmm, but you are right. will rework